File: backend/BookShelf.Application/DTOs/StatisticsDto.cs
namespace BookShelf.Application.DTOs;

public class StatisticsDto
{
    public int TotalBooks { get; set; }
    public int WantToRead { get; set; }
    public int CurrentlyReading { get; set; }
    public int Finished { get; set; }
    public int BooksReadThisYear { get; set; }
    public Dictionary<string, int> GenreDistribution { get; set; } = [];
}

File: mobile/app/src/main/java/com/j0nathan550/bookshelf/data/remote/dto/StatisticsDto.kt
package com.j0nathan550.bookshelf.data.remote.dto

data class StatisticsDto(
    val totalBooks: Int,
    val wantToRead: Int,
    val currentlyReading: Int,
    val finished: Int,
    val booksReadThisYear: Int,
    val genreDistribution: Map<String, Int> = emptyMap(),
)
